Tests/SonarScanner.MSBuild.PreProcessor.Test/PreProcessorTests.cs

```diff
@@ -122,9 +122,36 @@ public void PreProc_EndToEnd_SuccessCase()
             mockServer.AssertMethodCalled("TryGetQualityProfile", 2); // C# and VBNet
             mockServer.AssertMethodCalled("GetRules", 2); // C# and VBNet
 
+            logger.DebugMessages.Should().Contain("Base branch parameter was not provided. Incremental PR analysis is disabled.");
+
             AssertAnalysisConfig(settings.AnalysisConfigFilePath, 2, logger);
         }
 
+        [TestMethod]
+        public async Task PreProc_WithPullRequestBranch()
+        {
+            const string baseBranch = "BASE_BRANCH";
+            var workingDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var logger = new TestLogger();
+            var mockServer = new MockSonarQubeServer();
+            mockServer.Data.Languages.Add("cs");
+
+            var mockFactory = new MockObjectFactory(mockServer, Mock.Of<ITargetsInstaller>(), Mock.Of<IAnalyzerProvider>());
+
+            using (PreprocessTestUtils.CreateValidNonTeamBuildScope())
+            using (new WorkingDirectoryScope(workingDir))
+            {
+                var preProcessor = new TeamBuildPreProcessor(mockFactory, logger);
+
+                // Act
+                var args = CreateArgs("key", "name", new Dictionary<string, string> { { SonarProperties.PullRequestBase, baseBranch } }).ToArray();
```

Comment:
This will be a rebase conflict on arguments, right?

Reply:
No, the other PR is on top.